dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -131,6 +131,11 @@ def __init__(self, exec_func, **kwargs):
         self._cur_subpipeline_id = -1
         self._exec_func = exec_func
 
+        import numpy as np
+        if 'seed' in kwargs:
+            np.random.seed(kwargs['seed'])
+        self._seeds = np.random.randint(-2**63, 2**63-1, 1024)
```

Comment:
Doesn't this reseed the global state? I think such usage is discouraged. 
As an alternative we can create a new random generator, seeding it with user provided seed: https://numpy.org/doc/stable/reference/random/generator.html#numpy.random.default_rng

In that case we probably don't need to keep the list of initial seeds (up to you), we can run our private generator on demand.
Also, I don't know how DALI works when you pass negative seeds to it, but I think -1 might have a special meaning. I would suggest checking that or sticking to non-negative or positive range.

Reply:
done

Reply:
I tried to imitate seed generation from backend, which uses fixed size list of random values that can also be negative.

Reply:
done